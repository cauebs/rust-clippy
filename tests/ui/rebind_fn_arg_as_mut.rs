// edition:2018

#![warn(clippy::rebind_fn_arg_as_mut)]

fn f(x: bool) {
    let mut x = x;
}

trait T {
    fn tm1(x: bool) {
        let mut x = x;
    }
    fn tm2(x: bool);
}

struct S;

impl S {
    fn m(x: bool) {
        let mut x = x;
    }
}

impl T for S {
    fn tm2(x: bool) {
        let mut x = x;
    }
}

fn f_no_lint(mut x: bool) {
    let mut x = x;
}

async fn expansion<T>(_: T) {}

fn main() {}
