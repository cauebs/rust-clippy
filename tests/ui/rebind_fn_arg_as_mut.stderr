error: argument `x` is declared as not mutable, and later rebound as mutable
  --> $DIR/rebind_fn_arg_as_mut.rs:5:6
   |
LL | fn f(x: bool) {
   |      ^ help: consider just declaring as mutable: `mut x`
   |
   = note: `-D clippy::rebind-fn-arg-as-mut` implied by `-D warnings`
help: and remove this
  --> $DIR/rebind_fn_arg_as_mut.rs:6:5
   |
LL |     let mut x = x;
   |     ^^^^^^^^^^^^^^

error: argument `x` is declared as not mutable, and later rebound as mutable
  --> $DIR/rebind_fn_arg_as_mut.rs:10:12
   |
LL |     fn tm1(x: bool) {
   |            ^ help: consider just declaring as mutable: `mut x`
   |
help: and remove this
  --> $DIR/rebind_fn_arg_as_mut.rs:11:9
   |
LL |         let mut x = x;
   |         ^^^^^^^^^^^^^^

error: argument `x` is declared as not mutable, and later rebound as mutable
  --> $DIR/rebind_fn_arg_as_mut.rs:19:10
   |
LL |     fn m(x: bool) {
   |          ^ help: consider just declaring as mutable: `mut x`
   |
help: and remove this
  --> $DIR/rebind_fn_arg_as_mut.rs:20:9
   |
LL |         let mut x = x;
   |         ^^^^^^^^^^^^^^

error: argument `x` is declared as not mutable, and later rebound as mutable
  --> $DIR/rebind_fn_arg_as_mut.rs:25:12
   |
LL |     fn tm2(x: bool) {
   |            ^ help: consider just declaring as mutable: `mut x`
   |
help: and remove this
  --> $DIR/rebind_fn_arg_as_mut.rs:26:9
   |
LL |         let mut x = x;
   |         ^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

